recipes/leptonica/all/conanfile.py

```diff
@@ -0,0 +1,114 @@
+            self.requires.add("openjpeg/2.3.1")
+        if self.options.with_webp:
+            self.requires.add("libwebp/1.0.3")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.remove("fPIC")
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+        # Leptonica is very sensitive to package layout
+        # so use source directory same as _source_subfolder
+        os.rename(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                  os.path.join(self._source_subfolder, "CMakeListsOriginal.txt"))
+        shutil.copy("CMakeLists.txt",
+                    os.path.join(self._source_subfolder, "CMakeLists.txt"))
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions['STATIC'] = not self.options.shared
+        cmake.definitions['BUILD_PROG'] = False
+        # avoid finding system libs
+        cmake.definitions['CMAKE_DISABLE_FIND_PACKAGE_GIF'] = not self.options.with_gif
+        cmake.definitions['CMAKE_DISABLE_FIND_PACKAGE_PNG'] = not self.options.with_png
+        cmake.definitions['CMAKE_DISABLE_FIND_PACKAGE_TIFF'] = not self.options.with_tiff
+        cmake.definitions['CMAKE_DISABLE_FIND_PACKAGE_JPEG'] = not self.options.with_jpeg
+        cmake.definitions['CMAKE_DISABLE_FIND_PACKAGE_webp'] = not self.options.with_webp
+        cmake.definitions['CMAKE_DISABLE_FIND_PACKAGE_openjp2'] = not self.options.with_openjpeg
+
+        cmake.configure(source_folder=self._source_subfolder)
+        return cmake
+
+    def build(self):
+        # disable pkgconfig
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeListsOriginal.txt"),
+                              "if (PKG_CONFIG_FOUND)",
+                              "if (FALSE)")
+
+        # upstream uses obsolete FOO_LIBRARY that is not generated
+        # by cmake_find_package generator (upstream PR 456)
+        for dep in ('GIF', 'TIFF', 'PNG', 'JPEG', 'ZLIB'):
+            tools.replace_in_file(os.path.join(self._source_subfolder, "src", "CMakeLists.txt"),
+                                  dep + "_LIBRARY",
+                                  dep + "_LIBRARIES")
+
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        self.copy(pattern="leptonica-license.txt", dst="licenses", src=self._source_subfolder)
+        # remove pkgconfig
+        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
+        # remove cmake
+        tools.rmdir(os.path.join(self.package_folder, 'cmake'))
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
```

Comment:
```suggestion
        self.cpp_info.libs = tools.collect_libs(self)
        if self.settings.os == "Linux":
            self.cpp_info.system_libs = ["m"]
```

ldd /home/uilian/.conan/data/leptonica/1.78.0/_/_/package/93c846fd347a6ec336d737dee0587dc940b039f7/lib/libleptonica.so
	linux-vdso.so.1 (0x00007fffd3dca000)
	libm.so.6 => /usr/lib/libm.so.6 (0x00007f5eb0ffe000)
	libc.so.6 => /usr/lib/libc.so.6 (0x00007f5eb0e37000)
	/usr/lib64/ld-linux-x86-64.so.2 (0x00007f5eb161a000)